iceoryx_posh/test/moduletests/test_capro_service.cpp

```diff
@@ -114,32 +137,69 @@ TEST_F(ServiceDescription_test, CtorNoParams)
+    ServiceDescription csd1 = ServiceDescription(testService1, testInstance1, testEvent1, testHash);
+    ServiceDescription csd2 = ServiceDescription(testService2, testInstance2, testEvent2, testHash);
+    EXPECT_THAT(csd1.getServiceIDString(), StrEq("Service"));
+    EXPECT_THAT(csd1.getInstanceIDString(), StrEq("Instance"));
+    EXPECT_THAT(csd1.getEventIDString(), StrEq("Event"));
+    EXPECT_EQ(uint32_t(1), csd1.getClassHash()[0]);
+    EXPECT_EQ(uint32_t(2), csd1.getClassHash()[1]);
+    EXPECT_EQ(uint32_t(3), csd1.getClassHash()[2]);
+    EXPECT_EQ(uint32_t(4), csd1.getClassHash()[3]);
+    EXPECT_THAT(csd2.getServiceIDString(), StrEq("65536"));
+    EXPECT_THAT(csd2.getInstanceIDString(), StrEq("65536"));
+    EXPECT_THAT(csd2.getEventIDString(), StrEq("65536"));
+    EXPECT_EQ(uint32_t(1), csd2.getClassHash()[0]);
+    EXPECT_EQ(uint32_t(2), csd2.getClassHash()[1]);
+    EXPECT_EQ(uint32_t(3), csd2.getClassHash()[2]);
+    EXPECT_EQ(uint32_t(4), csd2.getClassHash()[3]);
 }
 
 TEST_F(ServiceDescription_test, operatorEq)
 {
-    EXPECT_TRUE(csd1 == csd1Eq);
-    EXPECT_TRUE(csd2 == csd2Eq);
-    EXPECT_TRUE(csd3 == csd3Eq);
-    EXPECT_TRUE(csd4 == csd4Eq);
+    ServiceDescription::ClassHash testHash = {1, 2, 3, 4};
+    auto serialObj1 = iox::cxx::Serialization::create("1",
+                                                     instance1.c_str(),
+                                                     event1.c_str(),
+                                                     1,
+                                                     3,
+                                                     2,
+                                                     testHash[0],
+                                                     testHash[1],
+                                                     testHash[2],
+                                                     testHash[3],
+                                                     true);
+    auto serialObj2 = iox::cxx::Serialization::create("1",
+                                                     "3",
+                                                     event1.c_str(),
+                                                     1,
+                                                     3,
+                                                     2,
+                                                     testHash[0],
+                                                     testHash[1],
+                                                     testHash[2],
+                                                     testHash[3],
+                                                     true);   
+    ServiceDescription csd10 = ServiceDescription(serialObj1);                                              
+    ServiceDescription csd11 = ServiceDescription(serialObj2);
 
-    EXPECT_FALSE(csd1 == csd1Ne);
-    EXPECT_FALSE(csd2 == csd2Ne);
-    EXPECT_FALSE(csd3 == csd3Ne);
-    EXPECT_FALSE(csd4 == csd4Ne);
+    EXPECT_TRUE(csd6 == csd6Eq);
```

Comment:
Every `EXPECT_TRUE` and `EXPECT_FALSE` should be a single test with an expressive name. This is  hard to understand, error prone and hard to maintain.
A name suggestion would be `TwoServiceDescriptionsWithIdenticalServiceInstanceEventAreEqual`
or `TwoServiceDescriptionWhereEventDiffersAreNotEqual`

This is also helping the reviewer to check the tests for completeness.

Please follow this rule for every test case in here.